server.js

```diff
@@ -0,0 +1,16 @@
+var webpack = require('webpack')
+var WebpackDevServer = require('webpack-dev-server')
+var config = require('./webpack.config')
+
+new WebpackDevServer(webpack(config), {
+  publicPath: config.output.publicPath,
+  hot: true,
+  inline: true,
+  historyApiFallback: true,
+  noInfo: true
+}).listen(3000, '0.0.0.0', function(err, result) {
+  if (err) {
+    console.log(err)
+  }
+  console.log('Listening at 0.0.0.0:3000')
+})
```

Comment:
For the record, I'm not a huge fan of this approach where you basically run two servers in dev. I think it makes things overly complicated for the dev and introduces unhelpful differences between dev and prod environments.

But I've seen it done before, so maybe it's just a matter of opinion. Just be careful about hard-coding references to port `3000` anywhere in your actual application code because that will break on prod.